pkg/process/procutil/process_linux.go

```diff
@@ -53,34 +53,37 @@ type statInfo struct {
 	cpuStat    *CPUTimesStat
 }
 
-// Option is config options callback for system-probe
-type Option func(p *Probe)
-
 // WithReturnZeroPermStats configures whether StatsWithPermByPID() returns StatsWithPerm that
 // has zero values on all fields
 func WithReturnZeroPermStats(enabled bool) Option {
-	return func(p *Probe) {
-		p.returnZeroPermStats = enabled
+	return func(p Probe) {
+		if linuxProbe, ok := p.(*probe); ok {
+			linuxProbe.returnZeroPermStats = enabled
+		}
 	}
 }
 
 // WithPermission configures if process collection should fetch fields
 // that require elevated permission or not
 func WithPermission(enabled bool) Option {
-	return func(p *Probe) {
-		p.withPermission = enabled
+	return func(p Probe) {
+		if linuxProbe, ok := p.(*probe); ok {
+			linuxProbe.withPermission = enabled
+		}
 	}
 }
 
 // WithBootTimeRefreshInterval configures the boot time refresh interval
 func WithBootTimeRefreshInterval(bootTimeRefreshInterval time.Duration) Option {
-	return func(p *Probe) {
-		p.bootTimeRefreshInterval = bootTimeRefreshInterval
+	return func(p Probe) {
+		if linuxProbe, ok := p.(*probe); ok {
+			linuxProbe.bootTimeRefreshInterval = bootTimeRefreshInterval
+		}
 	}
 }
 
-// Probe is a service that fetches process related info on current host
-type Probe struct {
+// probe is a service that fetches process related info on current host
+type probe struct {
```

Comment:
I think we should probably rename this to something more meaningful.

Reply:
For the sake of keeping this PR smaller probably worth splitting that, though I think pondering a good alternative might take time - collector (taken), enumerator (odd), if you have a suggestion feel free to post here.